src/data/fintype/basic.lean

```diff
@@ -1228,6 +1228,20 @@ finset.subtype.fintype s
 @[simp] lemma fintype.card_coe (s : finset α) [fintype s] :
   fintype.card s = s.card := fintype.card_of_finset' s (λ _, iff.rfl)
 
+/-- Noncomputable equivalence between a finset `s` as a fintype and `fin (s.card)`. -/
+noncomputable def finset.equiv_fin (s : finset α) : s ≃ fin (s.card) :=
+fintype.equiv_fin_of_card_eq (fintype.card_coe _)
+
+/-- Noncomputable equivalence between a finset `s` as a fintype and `fin n`, when there is a
+proof that `s.card = n`. -/
+noncomputable def finset.equiv_fin_of_card_eq {s : finset α} {n : ℕ} (h : s.card = n) : s ≃ fin n :=
+fintype.equiv_fin_of_card_eq (h ▸ fintype.card_coe _)
```

Comment:
What you wrote is fine here, but note that `▸` uses a more complicated elaboration scheme. If you wanted to avoid it (and there's no need to) I'm just mentioning that this would work too:
```suggestion
fintype.equiv_fin_of_card_eq ((fintype.card_coe _).trans h)
```